odoo/exceptions.py

```diff
@@ -107,3 +110,38 @@ def __init__(self, msg, tb):
 
 class QWebException(Exception):
     pass
+
+
+class ViewError(except_orm, ValueError):
+
+    def __init__(self, msg, view):
+        vals = [self, msg, '']
+
+        if not view:
+            # an error occurred when loading a default view, find out especifically which one broke
+            loaded_view = view.env.context.get('loaded_view', False)
+            if loaded_view:
+                model = loaded_view[0]
+
+                try:
+                    # reload the views to retrigger a ViewError, this time with debug information
+                    # to properly find the view id that causes the error
+                    view.env[model].with_context(inherit_branding=True).load_views(*loaded_view[1:])
+                except ViewError as e:
+                    vals = [self, e.name, e.value, e.actions]
```

Comment:
Why don't you reintroduce a private method like `view._raise_view_error(msg)` on `ir.ui.view` to:
- retry the view construction like above and simply reraise `e`, or
- log the error and raise a `ViewError` with the right action.

This would automatically factor out the logging part, and `ViewError` would have no actual code.